### PR TITLE
fix(chat): fix deleting selected prompts in search mode (Issue #1874)

### PR DESCRIPTION
**Pull request by Gimir**
**Description:**

- fix delete selected prompts in search mode
- apply same changes to conversations epic as well

Issues:

- Issue #1874 

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Gimir** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1883.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Gimir** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1883.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/store/prompts/prompts.epics.ts [hunk 1]
```diff
@@ -724,31 +723,31 @@ const deleteChosenPromptsEpic: AppEpic = (action$, state$) =>
...
-        ...promptIds.filter(
-          (id) =>
-            chosenFolderIds.some((folderId) => id.startsWith(folderId)) &&
-            doesEntityContainSearchTerm(
-              parsePromptApiKey(id) as unknown as ShareEntity,
-              searchTerm,
-            ),
+        ...chosenPromptIds,
+        ...searchedPromptIds.filter((id) =>
+          chosenFolderIds.some((folderId) => id.startsWith(folderId)),
```
Comment (IlyaBondar) [inline apps/chat/src/store/prompts/prompts.epics.ts:736]: I think we just need to replace `doesEntityContainSearchTerm` by `searchedPromptIds.includes(id)` and don't change other parts
```suggestion
       ...chosenPromptIds.filter((id) =>
          searchedPromptIds.includes(id),
        ),
        ...promptIds.filter(
          (id) =>
            chosenFolderIds.some((folderId) => id.startsWith(folderId)) &&
            searchedPromptIds.includes(id),
        ),
      ]);
      const filteredPrompts = prompts.filter(
        (prompt) => !deletedPromptIds.includes(prompt.id),
      );
```
and `if (promptIds.length) {` -> `if (searchedPromptIds.length) {`

File: apps/chat/src/store/conversations/conversations.epics.ts [hunk 2]
```diff
@@ -2670,33 +2663,32 @@ const deleteChosenConversationsEpic: AppEpic = (action$, state$) =>
...
-        ...conversationIds.filter(
-          (id) =>
-            chosenFolderIds.some((folderId) => id.startsWith(folderId)) &&
-            doesEntityContainSearchTerm(
-              parseConversationApiKey(id) as unknown as ShareEntity,
-              searchTerm,
-            ),
+        ...chosenConversationIds,
+        ...searchedConversationIds.filter((id) =>
+          chosenFolderIds.some((folderId) => id.startsWith(folderId)),
```
Comment (IlyaBondar) [inline apps/chat/src/store/conversations/conversations.epics.ts:2677]: the same

File: apps/chat/src/store/prompts/prompts.reducers.ts [hunk 3]
```diff
@@ -456,14 +456,18 @@ export const promptsSlice = createSlice({
...
-                    prompt.id.startsWith(folderId) &&
-                    !state.chosenPromptIds.includes(prompt.id) &&
-                    !state.chosenFolderIds.some((chosenFolderId) =>
-                      prompt.id.startsWith(chosenFolderId),
-                    ),
-                ),
+                !state.prompts
+                  .filter((prompt) =>
+                    doesEntityContainSearchTerm(prompt, state.searchTerm),
+                  )
```
Comment (IlyaBondar) [inline apps/chat/src/store/prompts/prompts.reducers.ts:466]: I suppose the same change should be done for conversations

**Comment by Gimir** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1883.nightly-test.deltixhub.io
>E2E tests status:

**Review by IlyaBondar — APPROVED** [review]

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1883.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1883.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Gimir** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1883.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1883.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1883.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1883.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1883.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1883.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by irinakartun — APPROVED** [review]

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1883.nightly-test.deltixhub.io
>E2E tests status: success

**Pull request merged**